packages/UI/src/components/ShareDialog/ShareDialog.tsx

```diff
@@ -0,0 +1,169 @@
+import {
+  Twitter,
+  Facebook,
+  Instagram,
+  Linkedin,
+  Link as LinkIcon,
+} from 'lucide-react';
+import {
+  FacebookShareButton,
+  TwitterShareButton,
+  LinkedinShareButton,
+  InstapaperShareButton,
+} from 'next-share';
+import { tv } from 'tailwind-variants';
+import Button from '../atoms/Button';
+import {
+  Dialog,
+  DialogTrigger,
+  DialogContent,
+  DialogHeader,
+  DialogFooter,
+  DialogTitle,
+} from './../Dialog';
+
+const ShareDialogStyles = tv(
+  {
+    slots: {
+      dialogHeader:
+        'flex h-6 w-[360px] items-center justify-center font-nunito-sans text-xl uppercase tracking-wider',
+      dialogFooter:
+        'flex w-[320px] flex-col items-center justify-end gap-4 font-nunito-sans',
+      dialogBody: 'flex w-[320px] items-center justify-evenly gap-4',
+      shareBtn: 'flex h-10 w-10 items-center justify-center rounded shadow-lg',
+    },
+    variants: {
+      screenPseudo: {
+        md: {
+          dialogHeader: 'w-[448px]',
+        },
+        lg: {},
+        xl: {},
+      },
+    },
+  },
+  {
+    responsiveVariants: ['md', 'lg', 'xl'],
+  }
+);
+
+export interface ShareDialogProps {
+  /*
+   * The header of the dialog. For example, "Idea" or "Project".
+   */
+  header: string;
+  /*
+   * The title of the shared content. Defaults to the current page title.
+   */
+  title?: string;
+  /*
+   * The description of the shared content. Defaults to the current page description.
+   */
+  description?: string;
+  /*
+   * The quote of the shared content. Defaults to the current page description.
+   */
+  quote?: string;
+  /*
+   * The hashtag to share. Defaults to the current page hashtag.
+   */
+  hashtag?: string;
+  /*
+   * The URL to share. Defaults to the current URL.
+   */
+  url?: string;
+  /*
+   * The button to trigger the dialog.
+   */
+  button?: JSX.Element;
+}
+/*
+ * https://www.figma.com/file/EwzuhhvTulvFRMvhTD5VAh/DL-Universal-Design-System?type=design&node-id=11488-39095&mode=design&t=D8fOpTHCs7PlLEic-0
+ */
+
+export function ShareDialog({
+  title,
+  header,
+  url = '',
+  quote,
+  description,
+  hashtag,
+  button,
+}: ShareDialogProps) {
+  const { dialogHeader, dialogFooter, dialogBody, shareBtn } =
+    ShareDialogStyles({
+      screenPseudo: {
+        md: 'md',
+        lg: 'lg',
+        xl: 'xl',
+      },
+    });
+
+  const copyToClipboard = () => {
+    alert('Link copied to clipboard!');
+    navigator.clipboard.writeText(url);
+  };
+
+  return (
+    <>
+      <Dialog>
+        <DialogTrigger asChild>
+          {button ?? (
+            <Button buttonType="primary" buttonSize="standard">
+              Share
+            </Button>
+          )}
+        </DialogTrigger>
+
+        <DialogContent>
+          <DialogHeader>
+            <h3 className={dialogHeader()}>
+              <DialogTitle>Share this {header}</DialogTitle>

```

Comment:
`header` should probably be open to be worded however the consumer wants rather than being limited to `Share this` then inserting their word.